7.3.0 causes a warning (and an error) in Framework projects

### Nuget Package

JsonSchema.Net

### Package Version

7.3.0

### Operating System

Windows

### .Net Target (if relevant, please specify the version in the description)

.Net Framework

### A clear and concise description of the bug.

Using 7.3.0 in .NET Framework projects will result in a MSB3277 warning. It does not occur in .NET (Core) projects. In some projects, a CS1705 error may appear (I don't know why).

### What did you expect?

There should be no errors or warnings.

### Please add test code or steps to reproduce the behavior.

A Project that cause MSB3277 warning: https://github.com/na1307/test/tree/main/JsonTest/JsonTest

A Project that also cause CS1705 error: https://github.com/na1307/SimpleSixtarScorecard/tree/dependabot_nuget_default-3132ca9f24/SimpleSixtarScorecard

### Is there any other information you'd like to share regarding this bug?

The warning is as follows:

```
C:\Program Files\dotnet\sdk\9.0.101\Microsoft.Common.CurrentVersion.targets(2413,5): warning MSB3277:
      Found conflicts between different versions of "System.Text.Json" that could not be resolved.
      There was a conflict between "System.Text.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" and "System.Text.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51".
          "System.Text.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was chosen because it was primary and "System.Text.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" was not.
          References which depend on "System.Text.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [D:\nugetcache\system.text.json\8.0.0\lib\net462\System.Text.Json.dll].
              D:\nugetcache\system.text.json\8.0.0\lib\net462\System.Text.Json.dll
                Project file item includes which caused reference "D:\nugetcache\system.text.json\8.0.0\lib\net462\System.Text.Json.dll".
                  D:\nugetcache\system.text.json\8.0.0\lib\net462\System.Text.Json.dll
          References which depend on or have been unified to "System.Text.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" [].
              D:\nugetcache\json.more.net\2.0.1.2\lib\netstandard2.0\Json.More.dll
                Project file item includes which caused reference "D:\nugetcache\json.more.net\2.0.1.2\lib\netstandard2.0\Json.More.dll".
                  D:\nugetcache\json.more.net\2.0.1.2\lib\netstandard2.0\Json.More.dll
                  D:\nugetcache\jsonpointer.net\5.0.0\lib\netstandard2.0\JsonPointer.Net.dll
                  D:\nugetcache\jsonschema.net\7.3.0\lib\netstandard2.0\JsonSchema.Net.dll
              D:\nugetcache\jsonpointer.net\5.0.0\lib\netstandard2.0\JsonPointer.Net.dll
                Project file item includes which caused reference "D:\nugetcache\jsonpointer.net\5.0.0\lib\netstandard2.0\JsonPointer.Net.dll".
                  D:\nugetcache\jsonpointer.net\5.0.0\lib\netstandard2.0\JsonPointer.Net.dll
                  D:\nugetcache\jsonschema.net\7.3.0\lib\netstandard2.0\JsonSchema.Net.dll
              D:\nugetcache\jsonschema.net\7.3.0\lib\netstandard2.0\JsonSchema.Net.dll
                Project file item includes which caused reference "D:\nugetcache\jsonschema.net\7.3.0\lib\netstandard2.0\JsonSchema.Net.dll".
                  D:\nugetcache\jsonschema.net\7.3.0\lib\netstandard2.0\JsonSchema.Net.dll
```

I'm using SDK 9.0 here, but it seems to appear regardless of SDK version.

The error can be found at [this link](https://github.com/na1307/SimpleSixtarScorecard/actions/runs/12516359872/job/34915330289).

### Code of Conduct & Contributor Guidelines

- [X] I agree to follow this project's Code of Conduct and Contribution Guidelines.

--- a/src/JsonSchema.Tests/DependencyChecks.cs
+++ b/src/JsonSchema.Tests/DependencyChecks.cs
@@ -1,0 +1,57 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Json.Pointer;
+using NUnit.Framework;
+
+namespace Json.Schema.Tests;
+
+public class DependencyChecks
+{
+	// already copied as part of library build
+	private const string NuspecFile = "nuspec/JsonSchema.Net.nuspec";
+
+	private static readonly string[] focusedAssemblies =
+	[
+		"Json.More",
+		"JsonPointer.Net"
+	];
+
+	[Test]
+	public void VerifyDependencyVersions()
+	{
+		_ = JsonPointer.Create("load"); // need to load the assembly when the other tests aren't run
+	
+		var nuspecLines = File.ReadAllLines(NuspecFile);
+		var nugetDependencies = nuspecLines
+			.Select(x => Regex.Match(x, @"id=""(?<lib>.*)"" version=""(?<version>\d*\.\d*\.\d*(\.\d*)?(-.*)?)"""))
+			.Where(x1 => x1.Success)
+			.ToDictionary(GetLibName, GetVersion);
+
+		var library = typeof(JsonSchema).Assembly;
+		var domain = AppDomain.CurrentDomain.GetAssemblies().OrderBy(x => x.FullName);
+		var libDependencies = library.GetReferencedAssemblies()
+			.Where(x => focusedAssemblies.Contains(x.Name))
+			.ToDictionary(x => x.Name, x => domain.First(a => a.GetName().Name == x.Name));
+
+		foreach (var (nugetDependency, nugetVersion) in nugetDependencies)
+		{
+			var libDependency = libDependencies[nugetDependency];
+			var libVersion = ((AssemblyInformationalVersionAttribute?)libDependency
+				.GetCustomAttributes(typeof(AssemblyInformationalVersionAttribute))
+				.SingleOrDefault())?.InformationalVersion;
+
+			Assert.That(libVersion, Does.StartWith(nugetVersion));
+		}
+	}
+
+	private string? GetLibName(Match match)
+	{
+		var pkgName = match.Groups["lib"].Value;
+		return focusedAssemblies.FirstOrDefault(pkgName.StartsWith);
+	}
+
+	private string GetVersion(Match match) => match.Groups["version"].Value;
+}

--- a/src/JsonSchema/JsonSchema.csproj
+++ b/src/JsonSchema/JsonSchema.csproj
@@ -17,8 +17,8 @@
     <IncludeSymbols>true</IncludeSymbols>
     <SymbolPackageFormat>snupkg</SymbolPackageFormat>
     <PackageId>JsonSchema.Net</PackageId>
-    <Version>7.3.0</Version>
-    <FileVersion>7.3.0</FileVersion>
+    <Version>7.3.1</Version>
+    <FileVersion>7.3.1</FileVersion>
     <AssemblyVersion>7.0.0.0</AssemblyVersion>
     <Authors>Greg Dennis</Authors>
     <Description>JSON Schema built on the System.Text.Json namespace</Description>

--- a/src/JsonSchema/nuspec/JsonSchema.Net.nuspec
+++ b/src/JsonSchema/nuspec/JsonSchema.Net.nuspec
@@ -13,7 +13,7 @@
     <tags>json-schema validation schema json</tags>
     <repository type="git" url="https://github.com/json-everything/json-everything" />
     <dependencies>
-      <dependency id="JsonPointer.Net" version="5.0.0" exclude="Build,Analyzers" />
+      <dependency id="JsonPointer.Net" version="5.1.0" exclude="Build,Analyzers" />
     </dependencies>
   </metadata>
   <files>

